### DISCUSSION TITLE
What is this error?  "WARN [http server] route GET /v1/publicip/ip is unprotected by default"

### DISCUSSION THREAD

**Discussion by aries223** [qdm12/gluetun#2493] [Q&A] [answered]
Can someone explain this warning please? 
I've looked up the control-server.md authentication doc, but still a little unclear as what to do..

```
2024-09-27T19:36:14Z INFO [http server] 200 GET /ip wrote 250B to xxx.xxx.xx.xxx:43560 in 36.367µs
2024-09-27T19:47:34Z WARN [http server] route GET /v1/publicip/ip is unprotected by default, please set up authentication following the documentation at https://github.com/qdm12/gluetun-wiki/setup/advanced/control-server.md#authentication since this will become no longer publicly accessible after release v3.40.
```

**Comment by Arelius-D** [comment]
@aries223 you got any respond?

I have something similar I think..
![image](https://github.com/user-attachments/assets/171e8426-8856-4c2a-990c-79a039ed6f0d)

also the link from the loggs doesn't work..
![image](https://github.com/user-attachments/assets/0f49217e-600b-49d3-87a3-515e98b977a8)

**Reply by frepke** [reply]
Try this link: [https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/control-server.md#authentication](https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/control-server.md#authentication)

**Reply by Arelius-D** [reply]
Thank you!

By any chance you know why one have to setup and have Control Server?
Also, is there is some other documentations which explains how to actually do it? I read the link you provided but can't say I understand what I'm supposed to do...

With these changes to Gluetun utility my tiny Pi5 setup basically broke or just doesn't function.... my Pi-Hole broke, qBittorrent also, SMB is stil fine... and I basically have to shutdown the system or disconnect it from the network, change DNS server in my router and only then I'm able  to keep myself connected to internet.. something is interfering so my router keeps disconnecting but according to my ISP provider the fault is not with them, which I kind of agree with since none of my devices (or router) disconnects/drop connection while the Pi5 is shut down/disconnected..

**Reply by aries223** [reply]
> Try this link: https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/control-server.md#authentication

Agreed, that write up is not very understandable...

**Reply by frepke** [reply]
> Thank you!
> 
> 
> 
> By any chance you know why one have to setup and have Control Server?
> 
> Also, is there is some other documentations which explains how to actually do it? I read the link you provided but can't say I understand what I'm supposed to do...
> 
> 
> 
> With these changes to Gluetun utility my tiny Pi5 setup basically broke or just doesn't function.... my Pi-Hole broke, qBittorrent also, SMB is stil fine... and I basically have to shutdown the system or disconnect it from the network, change DNS server in my router and only then I'm able  to keep myself connected to internet.. something is interfering so my router keeps disconnecting but according to my ISP provider the fault is not with them, which I kind of agree with since none of my devices (or router) disconnects/drop connection while the Pi5 is shut down/disconnected..

If you don't have any purpose for a control server, simply don't use.

**Reply by aries223** [reply]
@frepke 
> If you don't have any purpose for a control server, simply don't use.

The docs says this:
"⚠️ all routes will become private by default after the v3.40.0 release ⚠️"
...and the log warning is telling us to use it. If we dont need it, and its not required, why warn the we need to use it?

Can you please add a clarification to the Docs page that it is not a required configuration and that we can safely ignore the warning? 

If that is actually not the case, then please add a clarification in the doc page as to what the control server is, why we need to use it, and what will happen if we dont..

Thanks

**Reply by frepke** [reply]
> @frepke 
> 
> > If you don't have any purpose for a control server, simply don't use.
> 
> 
> 
> The docs says this:
> 
> "⚠️ all routes will become private by default after the v3.40.0 release ⚠️"
> 
> ...and the log warning is telling us to use it. If we dont need it, and its not required, why warn the we need to use it?
> 
> 
> 
> Can you please add a clarification to the Docs page that it is not a required configuration and that we can safely ignore the warning? 
> 
> 
> 
> If that is actually not the case, then please add a clarification in the doc page as to what the control server is, why we need to use it, and what will happen if we dont..
> 
> 
> 
> Thanks

No cannot, we have to wait for Quentin's reaction.

**Reply by Arelius-D** [reply]
Question is, will he even notice this?

**Reply by aries223** [reply]
@qdm12 
Please reveiew this thread..

**Reply by aries223** [reply]
Hello, some review on this error is needed please

**Reply by qdm12** [reply, marked as answer]
Hello there, sorry for the delay seeing this; I don't have notifications turned on due to the high number of issues/discussions and being the only maintainer.

> ...and the log warning is telling us to use it. If we dont need it, and its not required, why warn the we need to use it?

That warning is only logged IF you access the /v1/publicip/ip route so you are using the control server. The Wiki link is fixed in e93fd193b9e08076e6281f0995837ea1644d3598 - my bad, and thanks @frepke for pointing in the right direction!

> Can you please add a clarification to the Docs page that it is not a required configuration and that we can safely ignore the warning?
> If that is actually not the case, then please add a clarification in the doc page as to what the control server is, why we need to use it, and what will happen if we dont..

Re-read please. From next release, it will be compulsory and so you won't be able to access the /v1/publicip/ip route anymore without authentication. So you can't really safely ignore the warning unless you enjoy having your setup randomly broken after a Gluetun `v3` image upgrade.

Also I have no idea what is accessing that `/v1/publicip/ip` route, it must be one of the connected containers I guess. This is especially showcasing why authentication is important, given (for example) you don't actually know what connected container you have is using the Gluetun control server.
